### PR TITLE
Emit language metric after function indexing

### LINKED ISSUE #3826: Emit metric for language worker after jobhost initializes
We need to emit a log to FunctionsMetrics when the host starts up any language worker.

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #3826

### CODE REVIEW

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 1]
```diff
@@ -264,11 +264,6 @@ protected override async Task StartAsyncCore(CancellationToken cancellationToken
         public async Task InitializeAsync()
         {
             _stopwatch.Start();
-            if (!_environment.IsPlaceholderModeEnabled())
```
Comment (pragnagopa) [inline src/WebJobs.Script/Host/ScriptHost.cs]: analytics logs ignores placeholder sites. No need to check for placeholdermode

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/NodeEndToEndTests.cs [hunk 2]
```diff
@@ -231,7 +231,8 @@ public async Task Scenario_Logging()
             // verbose logs are filtered by default (the TestLogger explicitly
             // allows all levels for testing purposes)
             var key = MetricsEventManager.GetAggregateKey(MetricEventNames.FunctionUserLog, "Scenarios");
-            // Assert.Equal(9, Fixture.MetricsLogger.LoggedEvents.Where(p => p == key).Count());
+            //Assert.Equal(9, Fixture.MetricsLogger.LoggedEvents.Where(p => p == key).Count());
+            //Assert.Contains($"{MetricEventNames.HostStartupRuntimeLanguage}.node", Fixture.MetricsLogger.LoggedEvents);
```
Comment (pragnagopa) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/NodeEndToEndTests.cs]: @brettsam -  Any ideas how to ovverride metrics logger for these tests? Test has a TODO
```// TODO: Re-enable once we can override IMetricsLogger```

Comment (brettsam) [reply test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/NodeEndToEndTests.cs]: I have a PR out that enables this -- need to get some eyes on it though: https://github.com/Azure/azure-functions-host/pull/4109

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 3]
```diff
@@ -277,6 +272,10 @@ public async Task InitializeAsync()
                 // Generate Functions
                 IEnumerable<FunctionMetadata> functions = GetFunctionsMetadata();
 
+                _workerRuntime = _workerRuntime ?? Utility.GetWorkerRuntime(functions);
```
Comment (brettsam) [inline src/WebJobs.Script/Host/ScriptHost.cs]: Is the call to `GetWorkerRuntime()` the fix? I assume we were logging "none" a lot?

File: src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs [hunk 4]
```diff
@@ -94,19 +94,18 @@ public LanguageWorkerState CreateWorkerStateWithExistingChannel(string language,
...
             _languageWorkerChannelManager.ShutdownStandbyChannels(functions);
-            workerRuntime = workerRuntime ?? Utility.GetWorkerRuntime(functions);
 
             if (Utility.IsSupportedRuntime(workerRuntime, _workerConfigs))
             {
                 ILanguageWorkerChannel initializedChannel = _languageWorkerChannelManager.GetChannel(workerRuntime);
                 if (initializedChannel != null)
                 {
-                    _logger.LogDebug($"Found initialized language worker channel for runtime: {0}", workerRuntime);
+                    _logger.LogDebug("Found initialized language worker channel for runtime: {workerRuntime}", workerRuntime);
```
Comment (brettsam) [inline src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs]: Nice structured logging :-) 

Is this important to the fix, or just nice to have?

File: src/WebJobs.Script/Host/ScriptHost.cs (hunk 3 above)
Comment (pragnagopa) [reply src/WebJobs.Script/Host/ScriptHost.cs]: Yes. GetWorkerRuntime calculates language for an app if app setting "Functions_Worker_Runtime" is not set.

File: src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs (hunk 4 above)
Comment (pragnagopa) [reply src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs]: Yes i made a typo when I added these logs :)

**Pull request merged**